pkg/utils/sysfs.go

```diff
@@ -63,6 +67,22 @@ func SetCPUScalingMaxFreq(cpu ID, freq int) error {
 	return setCPUFreqValue(cpu, "scaling_max_freq", freq)
 }
 
+// SetCPUScalingGovernor sets the governor mode of a given CPU
+func SetCPUScalingGovernor(cpu ID, governor string) error {
+	return setCPUScalingGovernor(cpu, "scaling_governor", governor)
+}
+
+// SetCPUsScalingGovernor sets the scaling_governor of a given set of CPUs
+func SetCPUsScalingGovernor(cpus []ID, governor string) error {
+	for _, cpu := range cpus {
+		if err := SetCPUScalingGovernor(cpu, governor); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
nit: Just wondering if this function could/should try to be atomic ? IOW, if it should read out the previous settings and restore them if it fails mid-loop ?

Reply:
I think it is general a good idea and I can extend the code. Unfortunately os library doesn't provide any built in functionality for such cases meaning if we decide to cover that scenario then we will have steps like taking a backup of a file, then try to patch it and etc.

Reply:
Yes. you'd need to do a reading of the setting before you touch it in the loop, store it, and then try restoring the stored values on error branches. But IMO that's definitely a follow-up PR if we consider that worth doing...